packages/mermaid/src/diagrams/xychart/chartBuilder/TextDimensionCalculator.ts

```diff
@@ -0,0 +1,68 @@
+import { Dimension } from './Interfaces.js';
+
+export interface ITextDimensionCalculator {
+  getDimension(texts: string[], fontSize: number, fontFamily?: string): Dimension;
+}
+
+export class TextDimensionCalculator implements ITextDimensionCalculator {
+  getDimension(texts: string[], fontSize: number): Dimension {
+    return {
+      width: texts.reduce((acc, cur) => Math.max(cur.length, acc), 0) * fontSize,
+      height: fontSize,
+    };
+  }
+}
+
+export class TextDimensionCalculatorWithFont implements ITextDimensionCalculator {
```

Comment:
TextDimensionCalculator is not used anywhere, and we don't use `I` prefix for interfaces, so this can be simplified. 

```suggestion
export interface TextDimensionCalculator {
  getDimension(texts: string[], fontSize: number, fontFamily?: string): Dimension;
}

export class TextDimensionCalculatorWithFont implements TextDimensionCalculator {
```

Reply:
Was fixed, resolving